### ISSUE TITLE
Voetnoten in rapporten - voorstel tot automatische nummering

### ISSUE THREAD

**Issue by GeertBonte** [provinciesincijfers/PinC-taakopvolging#367]
In rapporten worden voetnoten momenteel handmatig genummerd. Waarom geen automatische nummering gebruiken, zoals nu al gebeurt voor de nummering van kaarten, tabellen en grafieken?

Dat zou het voordeel hebben dat de nummering van de voetnoten automatisch aangepast wordt wanneer een of meer voetnoten wegvallen in een conditionele sectie (net als bij kaarten, tabellen en grafieken). Ik weet niet of zo’n situatie zich voordoet in bestaande rapporten, maar ik had dit nodig in hoofdstuk 12 van de omgevingsanalyse.

Dit zou eenvoudig geïmplementeerd kunnen worden door volgende CSS-code toe te voegen aan het CSS-bestand `ipo_dynamic.css` (de eerste regel is een wijziging van een bestaande regel; overige regels zijn toegevoegd):

```
body { counter-reset: tabelnum 0 grafieknum 0 kaartnum 0 figuurnum 0 voetnootnum 0 tabelnumtekst 0 grafieknumtekst 0 kaartnumtekst 0 figuurnumtekst 0 voetnootnumtekst 0; }

.teller_voetnoot { counter-increment: voetnootnum 1; }
.teller_voetnoottekst { counter-increment: voetnootnumtekst 1; }
.teller_voetnoottekst::before { content: "zie voetnoot\00a0" counter(voetnootnumtekst); }
```

Volledig analoog aan de werkwijze voor kaarten, tabellen en grafieken worden er dus twee nieuwe tellers gebruikt: `voetnootnum` voor het nootnummer in de voetnoot en `voetnootnumtekst` voor het nummer van de bijbehorende nootverwijzing in de tekst; en ook twee nieuwe classes: `teller_voetnoot`, te gebruiken in de voetnoot, en `teller_voetnoottekst`, te gebruiken voor de verwijzing in de tekst.

In de tekst kun je dan als volgt naar een voetnoot verwijzen (vóór het nummer wordt automatisch ‘zie voetnoot ’ geplaatst, conform de [afspraken](https://provincies.incijfers.be/admin/jive/Report/Edit/sjabloon_grafiekstijl)):

```
... (<a href="##voetnoot1" id="voet1" class="teller_voetnoottekst"></a>)
```

In de voetnoot zelf gaat het dan als volgt:

```
<p><a href="##voet1" id="voetnoot1" class="teller_voetnoot"></a> Deze cijfers zijn niet beschikbaar op gemeenteniveau.</p>
```

Er is geen enkele noodzaak om een `span`-element te gebruiken: je kunt het `class`-attribuut evengoed bij het `a`-element plaatsen.

Mag ik voorstellen om deze aanpassing door te voeren in `ipo_dynamic.css`, zodat automatisch genummerde voetnoten in alle rapporten mogelijk worden? Het is daarbij NIET nodig om de bestaande rapporten aan te passen; de hard gecodeerde voetnootnummers in de huidige rapporten blijven gewoon werken. Dit schept enkel de mogelijkheid om in nieuwe rapporten voortaan de automatische nootnummering te gebruiken.

Eenmaal het CSS-bestand aangepast, verdient het wel aanbeveling om rapporten aan te passen waar bepaalde voetnoten wegvallen in conditionele secties (als die er al zouden zijn).

**Comment by joostschouppe** [comment]
De href ankers moet je wel nog steeds uniek maken begrijp ik. Wellicht dan minder verwarren om daarvoor dan `voet_blabla` en `link_blabla` te gebruiken, waarbij `blabla` een hint geeft van waarover het gaat in de voetnoot zelf. Om de nummering in orde te houden in het stuk met de voetnoten, moet je er dan wel nog altijd op letten dat de volgorde van de voetnoten hetzelfde is als die in de tekst. Zelfs dan lijkt me dit voor de programmeur wel handiger, want als er ergens een voet tussenkomt moet je toch nog altijd enkel maar de voet op de juiste plaats zetten, terwijl je anders heel de nummering moet herzien.

**Comment by GeertBonte** [comment]
Ja, uiteraard moet elk anker uniek zijn, en de noten onderaan moeten in dezelfde volgorde staan als de verwijzingen in de tekst. Ik zou zelf wel eerder geneigd zijn om niet `voet_blabla` te gebruiken maar nog steeds `voet1`, `voet2`, etc. Stel dat er achteraf nog een extra noot ingevoegd moet worden tussen 1 en 2 en je wilt niet alle ankers aanpassen, dan kun je voor de nieuwe noot als anker `voet1a` of `voet1b` gebruiken. De nootnummers zelf worden gegenereerd via CSS en zullen zich vanzelf aanpassen.

**Comment by joostschouppe** [comment]
Ik heb je stukje code toegevoegd aan IPO_dynamic. David zal het uitrollen in alle rapporten.
Concreet dus=
- alle verwijzingen naar voetnoten aanpassen naar iets als `<a href="##voetnoot1" id="voet1" class="teller_voetnoottekst"></a>`. Hierbij dus het equivalent van voetnoot1 en voet1 ongewijzigd laten, en een bestaande tekst van de link verwijderen.
- In het deel met de voetnoten enkel `class="teller_voetnoot"` toevoegen in alle dingen als `<a href="##voet1" id="voetnoot1" class="teller_voetnoot"></a>`

**Comment by GeertBonte** [comment]
> Ik heb je stukje code toegevoegd aan IPO_dynamic. David zal het uitrollen in alle rapporten. Concreet dus
> 
> * alle verwijzingen naar voetnoten aanpassen naar iets als `<a href="##voetnoot1" id="voet1" class="teller_voetnoottekst"></a>`. Hierbij dus het equivalent van voetnoot1 en voet1 ongewijzigd laten, en een bestaande tekst van de link verwijderen.
> * In het deel met de voetnoten enkel `class="teller_voetnoot"` toevoegen in alle dingen als `<a href="##voet1" id="voetnoot1" class="teller_voetnoot"></a>`

Volgens het [sjabloon](https://provincies.incijfers.be/databank/report/?id=sjabloon_rapport_vertrekproject) en het [sjabloon afspraken](https://provincies.incijfers.be/databank/report/?id=sjabloon_grafiekstijl) wordt in de tekst naar voetnoten verwezen met 'zie voetnoot 1', maar dat is niet in alle rapporten op die manier geïmplementeerd. In sommige rapporten ontbreekt de tekst 'zie voetnoot' en staat het nootnummer in superscript (bv. het [rapport adaptatiemaatregelen](https://provincies.incijfers.be/databank/report/?id=rapport_adaptatiemaatregelen)). In mijn voorstel wordt in de toegevoegde CSS-code de tekst 'zie voetnoot' automatisch gegenereerd, maar als dat niet gewenst is, kun je dat weglaten. Aan de andere kant zou het natuurlijk beter zijn dat in alle rapporten op dezelfde manier naar voetnoten verwezen wordt.

Misschien eerst even in de redactiegroep bespreken alvorens dit overal aan te passen?

**Comment by joostschouppe** [comment]
Geert, met jouw hulp staat het nu helemaal klaar. Ik heb het in beide sjablonen klaargezet. Op zich was het reeds duidelijk hoe het moet zijn. Het uitrollen van afspraken zoals ze in het sjabloon staan hoeft niet nog eens goedgekeurd te worden door de redactiegroep. David kan dus gewoon overal de bestaande verwijzingen verwijderen waar nodig en die superscripts doen vervallen.
Ik agendeer het wel "ter info" op de volgende redactiegroep.

**Comment by joostschouppe** [comment]
Redactiegroep 23/3/2023 - bevestiging uitrol

**Comment by davidcardinaels** [comment]
Ik heb het toegevoegd in alle rapporten.

**Comment by GeertBonte** [comment]
Nog dit aandachtspuntje: wanneer een nootverwijzing (in de tekst) enkel geplaatst wordt wanneer aan een bepaalde voorwaarde is voldaan, dan moet je erop toezien dat de voetnoot zelf ook alleen maar geplaatst wordt onder diezelfde voorwaarde. – Dat staat los van de automatische nummering, dat was voorheen ook al zo. Een voorbeeld (uit het rapport klimaatscenario’s):

```
#if Equal(code(Parent(item(input_geo,0), provincie)), "provincie_30000") or Equal(code(Parent(item(input_geo,1), provincie)), "provincie_30000") then#

<p>... (<a href="##voetnoot21" id="voet21" class="teller_voetnoottekst"></a>) ...</p>

#else#
#end if#
```

Bovenstaand stukje tekst wordt alleen maar geplaatst wanneer het rapport opgevraagd wordt voor een gemeente in de provincie West-Vlaanderen. In de andere gevallen valt de tekst weg en wordt er dus niet naar de voetnoot verwezen. In dat geval mag de bijbehorende voetnoot bijgevolg ook niet getoond worden. In de voetnoten moet daarom het volgende staan:

```
#if Equal(code(Parent(item(input_geo,0), provincie)), "provincie_30000") or Equal(code(Parent(item(input_geo,1), provincie)), "provincie_30000") then#

<p><a href="##voet21" id="voetnoot21" class="teller_voetnoot"></a> ...</p>

#else#
#end if#
```

@StefanJacques: bovenstaand scenario deed zich voor bij voetnoten 21 en 22 in het rapport kimaatscenario’s, waar die voetnoten ten onrechte ook werden geplaatst wanneer er in de tekst niet naar verwezen werd. Ik ben zo vrij geweest dat aan te passen.
